### PR TITLE
F/597-start-multiple-workloads

### LINKED ISSUE #597: start multiple workloads
**Is your feature request related to a problem? Please describe:**  
It is possible to load more than one workload but not start multiple workloads.

**Please describe the solution you'd like:**  
By changing the radio buttons to checkboxes we will be able to select multiple workloads and start them.

**Affected Component(s):**  
`Frontend`

### PR DESCRIPTION
**Pull request by kathwill**
# Resolves #597 

**Does your pull request add a feature? Please describe:**  
It is now possible to start multiple workloads and start/stop of the workload and of the workers is separated.

**Affected Component(s):**  
`Frontend`

**Additional context:**  
Start multiple workloads via checkboxes and choose frequency for every loaded workload:
MultipleWorkloads7

Customize weights for multiple workloads:
MultipleWorkloads3
MultipleWorkloads4

### CODE REVIEW

File: hyrisecockpit/frontend/src/components/workload/FrequencyHandler.vue [hunk 1]
```diff
@@ -1,24 +1,30 @@
...
-      dense
-      @change="$emit('change', frequency)"
-    ></v-text-field>
-  </span>
+  <div>
+    <div v-for="(frequency, index) in frequencies" :key="index">
+      <v-row>
+        <v-slider
+          class="frequency-slider"
+          v-model="frequencies[index]"
```
Comment (schTi) [inline hyrisecockpit/frontend/src/components/workload/FrequencyHandler.vue]: please change the order of those statements

File: hyrisecockpit/frontend/src/components/workload/FrequencyHandler.vue [hunk 2]
```diff
@@ -1,24 +1,30 @@
...
+          max="1000"
+          :disabled="
+            !loadedWorkloads.includes(availableWorkloads[index]) || disabled
+          "
+          @click="$emit('change', index, frequency)"
+          @end="$emit('change', index, frequency)"
+        ></v-slider>
+        <v-text-field
+          class="frequency-text-field"
+          v-model="frequencies[index]"
```
Comment (schTi) [inline hyrisecockpit/frontend/src/components/workload/FrequencyHandler.vue]: same here

File: hyrisecockpit/frontend/src/components/workload/FrequencyHandler.vue [hunk 3]
```diff
@@ -28,34 +34,60 @@ import {
...
 
 export default defineComponent({
   name: "FrequencyHandler",
   props: {
-    initialFrequency: {
-      type: Number,
-      default: 0,
+    initialFrequencies: {
+      type: Array,
+      default: [],
```
Comment (schTi) [inline hyrisecockpit/frontend/src/components/workload/FrequencyHandler.vue]: Thats not gonna work, for complex types like objects or arrays you need to use an initializer. Basically this means you need to change the default value to a function like this: `() => []`

File: hyrisecockpit/frontend/src/components/workload/WorkloadGeneration.vue [hunk 4]
```diff
@@ -158,40 +172,28 @@ export default defineComponent({
   },
   setup(props: {}, context: SetupContext): Data {
     const tab = ref<number>(0);
-    const workloadHandler = useWorkloadHandler();
+    const workloadAction = useWorkloadAction(context);
```
Comment (schTi) [inline hyrisecockpit/frontend/src/components/workload/WorkloadGeneration.vue]: In my opinion there are multiple different actions, so it should be named `workloadActions` and `useWorkloadActions`

File: hyrisecockpit/frontend/src/components/workload/WorkloadGeneration.vue [hunk 5]
```diff
@@ -214,25 +216,50 @@ function useWorkloadAction(
...
+      initialiseWorkloadSelector(response.data);
+    }
+    getLoadedWorkloadData().then((response: any) => {
+      if (response.data.length > 0) {
+        initialiseWorkloadActions(response.data);
+      }
+    });
+  });
+
+  function initialiseWorkloadSelector(workloads: any): void {
```
Comment (schTi) [inline hyrisecockpit/frontend/src/components/workload/WorkloadGeneration.vue]: I dont know but `initialise` looks weird to me, I am only familiar with `initialize`

File: hyrisecockpit/frontend/src/components/workload/WorkloadGeneration.vue [hunk 6]
```diff
@@ -346,13 +397,19 @@ function useWorkloadDataHandler(context: SetupContext): WorkloadDataHandler {
   const { emitDatabaseStatusChangedEvent } = useDatabaseEvents();
 
   function handleWorkloadDataChange(workload: Workload): void {
-    workloadData[workload].loading = true;
+    loadingWorkloads.value.push(workload);
     changeWorkloadData = false;
-    if (!workloadData[workload].loaded) {
+    if (!loadedWorkloads.value.includes(workload)) {
+      //load workload data
```
Comment (schTi) [inline hyrisecockpit/frontend/src/components/workload/WorkloadGeneration.vue]:
```suggestion
      // load workload data
```

**Review by schTi — CHANGES_REQUESTED** [review]
Looks great, just some really small issues

File: hyrisecockpit/frontend/src/components/workload/WorkloadGeneration.vue (hunk 5 above)
Comment (kathwill) [reply hyrisecockpit/frontend/src/components/workload/WorkloadGeneration.vue]: It's British English, but I will change it.

File: hyrisecockpit/frontend/src/components/workload/FrequencyHandler.vue (hunk 3 above)
Comment (kathwill) [reply hyrisecockpit/frontend/src/components/workload/FrequencyHandler.vue]: I think it's only a naming issue then. I wanted to make clear that the prop sets the initial frequencies (which are initialised in the WorkloadGeneration) and there are then changed by using the sliders or text fields of the FrequencyHandler

**Review by schTi — APPROVED** [review]
Thanks for the small fixes. If you can ensure that everything is working, feel free to merge. Good job

**Pull request merged**